Detectors/FIT/FDD/simulation/src/Digits2Raw.cxx

```diff
@@ -0,0 +1,173 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+//
+//  RAW data format - DataFormat/Detectors/FIT/FDD/RawEventData
+
+#include "FDDBase/Constants.h"
+#include "FDDSimulation/Digits2Raw.h"
+#include <TTree.h>
+#include <cassert>
+
+using namespace o2::fdd;
+
+ClassImp(Digits2Raw);
+
+//_____________________________________________________________________________________
+void Digits2Raw::readDigits(const std::string& outDir, const std::string& fileDigitsName)
+{
+  LOG(INFO) << "==============FDD: Digits2Raw::convertDigits" << std::endl;
+  LookUpTable lut(true);
+
+  std::string outd = outDir;
+  if (outd.back() != '/') {
+    outd += '/';
+  }
+
+  // Register PM links linearly
+  for (int iPmLink = 0; iPmLink < Nmodules; ++iPmLink) {
+    uint64_t feeId = uint64_t(iPmLink);
+    uint32_t linkId = uint32_t(iPmLink);
```

Comment:
feeId is actually ``uint16_t`` and the linkId is ``uint8_t``, see 
``RawFileWriter::registerLink(uint16_t fee, uint16_t cru, uint8_t link, uint8_t endpoint, std::string_view outFileNameV)``